pkg/instrumentation/javaagent.go

```diff
@@ -50,12 +50,18 @@ func injectJavaagent(logger logr.Logger, javaSpec v1alpha1.Java, pod corev1.Pod,
 			logger.Info("Skipping javaagent injection, the container defines JAVA_TOOL_OPTIONS env var value via ValueFrom", "container", container.Name)
 			return pod
 		}
-		container.Env[idx].Value = container.Env[idx].Value + javaJVMArgument
+
+		if IsEnvVarValueInstrumentationMissing(container.Env[idx], javaJVMArgument) {
```

Comment:
Why is this check needed? 

The `IsAutoInstrumentationInjected` is not enough?